src/metabase/upload.clj

```diff
@@ -117,9 +117,26 @@
   [n values]
   (first (partition n n (repeat nil) values)))
 
+(defn- normalize-column-name
+  [[raw-name index]]
+  (let [normalized (u/slugify (str/trim raw-name))]
+    (if-not (str/blank? normalized)
+      normalized
+      (format "unnamed_column_%s" (inc index)))))
```

Comment:
`unnamed_column` should use the same deduplicate logic as normal columns. So you could just call it `unnamed_column` here and dedupe it later.